### DISCUSSION TITLE
请问“暑期集训保留体力”是怎么实现的

### DISCUSSION THREAD

**Discussion by sixZI** [NateScarlet/auto-derby#199] [Q&A] [answered]
十分好奇是如何实现的，我自己没能找到相关代码，如果可以的话能指点一下代码所处的地方吗？

**Comment by NateScarlet** [comment, marked as answer]
https://github.com/NateScarlet/auto-derby/blob/7abde423fffb532fa6195e2202212cc603351486/auto_derby/single_mode/training/training_score.py#L60-L64

https://github.com/NateScarlet/auto-derby/blob/7abde423fffb532fa6195e2202212cc603351486/auto_derby/single_mode/commands/rest.py#L25-L28

https://github.com/NateScarlet/auto-derby/blob/7abde423fffb532fa6195e2202212cc603351486/auto_derby/single_mode/go_out.py#L128-L130

**Reply by sixZI** [reply]
感谢解答！ 
vit = max(min(trn.vitality, 1 - ctx.vitality), 0) * ctx.max_vitality * 0.6 
 if ctx.date[1:] in ((6, 1),): 
     vit *= 1.2 
 if ctx.date[1:] in ((6, 2), (7, 1), (7, 2), (8, 1)): 
     vit *= 1.5 
这一段的意思是否是在6月前半体力值的分数为原来的1.2倍，在夏训期间分数为原来1.5倍？

**Reply by NateScarlet** [reply]
对

**Reply by sixZI** [reply]
请问如果想写个提升夏休前更倾向于恢复体力的插件，以下写法是正确的吗？
class Training(single_mode.Training):
    def score(self, ctx: single_mode.Context) -> float:
        ret = super().score(ctx)
        if ctx.date[1:] in ((6, 1),(6, 2)):
            ret += self.vitality * 0.3
        return ret

**Reply by NateScarlet** [reply]
逻辑上应该没错 效果你可以用 pause_before_command 插件来测试 日志里面会显示所有命令的评分

**Reply by sixZI** [reply]
好的，感谢回复（回复速度好快
